taichi/transforms/constant_fold.cpp

```diff
@@ -177,11 +177,32 @@ class ConstantFold : public BasicStmtVisitor {
       auto evaluated =
           Stmt::make<ConstStmt>(LaneAttribute<TypedConstant>(new_constant));
       stmt->replace_with(evaluated.get());
-      modifier.insert_before(stmt, VecStatement(std::move(evaluated)));
+      modifier.insert_before(stmt, std::move(evaluated));
       modifier.erase(stmt);
     }
   }
 
+  void visit(BitExtractStmt *stmt) override {
+    auto input = stmt->input->cast<ConstStmt>();
+    if (!input)
+      return;
+    if (stmt->width() != 1)
+      return;
+    int32 result;
+    if (is_signed(input->val[0].dt)) {
+      result = (input->val[0].val_int() >> stmt->bit_begin) &
+               ((1LL << (stmt->bit_end - stmt->bit_begin)) - 1);
+    } else {
+      result = (input->val[0].val_uint() >> stmt->bit_begin) &
+               ((1LL << (stmt->bit_end - stmt->bit_begin)) - 1);
```

Comment:
I thought BitExtractStmt is only used for i32?

Reply:
We probably need to support unsigned inputs as well :-)